barretenberg/cpp/src/barretenberg/ecc/fields/field_docs.md

```diff
@@ -0,0 +1,190 @@
+Prime field documentation    {#field_docs}
+===
+Barretenberg has its own implementation of finite field arithmetic. The implementation targets 254 (bn254, grumpkin) and 256-bit (secp256k1, secp256r1) fields. Internally the field is representate as a little-endian C-array of 4 uint64_t limbs.
+
+## Field arithmetic
+### Introduction to Montgomery form {#field_docs_montgomery_explainer}
+We use Montgomery reduction to speed up field multiplication. For an original element  \f$ a ∈ F_p\f$ the element is represented internally as $$ a⋅R\ mod\ p$$ where \f$R = 2^d\ mod\ p\f$. The chosen \f$d\f$ depends on the build configuration:
+1. \f$d=29⋅9=261\f$ for builds that don't support the uint128_t type, for example, for WASM build
+2. \f$d=64⋅4=256\f$ for standard builds (x86_64).
+
+The goal of using Montgomery form is to avoid heavy division modulo \f$p\f$. To compute a representative of element $$c = a⋅b\ mod\ p$$ we compute $$c⋅R = (a⋅R)⋅(b⋅R) / R\ mod\ p$$, but we use an efficient division trick to avoid straight modular division. Let's look into the standard 4⋅64 case:
```

Comment:
what do you mean by representative?

Reply:
Internally an element c is represented by  $c \cdot R$